### ISSUE TITLE
New QuoteConsistencyBear for checking if the same string delimiter is used across the function

### ISSUE THREAD

**Issue by underyx** [coala-analyzer/coala-bears#557]
When the quote mark is set to `'`

- `'hi!'` should pass
- `"hi!"` should fail and be corrected to `'hi!'`
- `"'ello!"` should pass
- `'\'ello!'` should pass or be corrected to `"'ello!"` depending on a config value

**Comment by sils1297** [comment]
we have some annotationbear that alreay determines strings in the source code, that can be used as a dependency

**Comment by hypothesist** [comment]
@underyx How about `"Jack's got 'em"`? I understand that's a really tough, corner case.

**Comment by underyx** [comment]
@hypothesist how is it different from the third example:

>- "'ello!" should pass

?

If the question is about difficulty in the string parsing, I think the apostrophes can be totally ignored and not considered string delimiters if we've already seen an opening double quote.

**Comment by Makman2** [comment]
I would apply the same rules as python does:
1. Prefer single quotes
2. If a string contains single quotes only and no double quotes, use double quotes
3. If a string contains both, single and double quotes, prefer again single quotes and escape single quotes in string

**Comment by underyx** [comment]
@Makman2 yes, this should be one of the possible behaviors, but there should be config options for

1. prefer **single/double** quotes

and

2. If a string contains single quotes only and no double quotes, **use double quotes/escape single quotes**

**Comment by Makman2** [comment]
ah okay  :+1: :)

**Comment by AbdealiJK** [comment]
Seems like we had this discussion and after 2 weeks had the exact same discussion on https://github.com/coala-analyzer/coala-bears/issues/697 :P So, refer that in case more info needed ^^

**Comment by kriti21** [comment]
What's the difference between this one and the existing `QuotesBear` ?

**Comment by Makman2** [comment]
Are all the cases in the issue description working properly? We should maybe add additional tests for them in case we don't test something similar.

If everything is really already done, we can close.
